slotting/src/Cardano/Slotting/Block.hs

```diff
@@ -12,14 +12,18 @@ import Cardano.Prelude (NoUnexpectedThunks)
 import Codec.Serialise (Serialise (..))
 import Data.Word (Word64)
 import GHC.Generics (Generic)
+import qualified Quiet
 
 -- | The 0-based index of the block in the blockchain.
 -- BlockNo is <= SlotNo and is only equal at slot N if there is a block
 -- for every slot where N <= SlotNo.
 newtype BlockNo = BlockNo {unBlockNo :: Word64}
-  deriving stock (Show, Eq, Ord, Generic)
+  deriving stock (Eq, Ord, Generic)
   deriving newtype (Enum, Bounded, Num, Serialise, NoUnexpectedThunks)
 
+instance Show BlockNo where
+  showsPrec = Quiet.qshowsPrec
```

Comment:
I would have used
```haskell
import Quiet (Quiet (..))

  ..
  deriving (Show) via (Quiet BlockNo)
```
Not a fan?

Reply:
Same response as https://github.com/input-output-hk/ouroboros-network/pull/1019#discussion_r427992547